### PR TITLE
Fix horizontal bar chart not drawing values and add unit tests

### PR DESCRIPTION
**Pull request by liuxuan30**
regards https://github.com/danielgindi/Charts/pull/3854
I found the condition seems wrong and duplicated.
add unit tests for horizontal bar charts

### CODE REVIEW

File: Tests/Charts/HorizontalBarChartTests.swift [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    {
+        var entries: [ChartDataEntry] = Array()
+        for (i, value) in values.enumerated()
+        {
+            entries.append(BarChartDataEntry(x: Double(i), y: value, icon: UIImage(named: "icon", in: Bundle(for: self.classForCoder), compatibleWith: nil)))
+        }
+        return entries
+    }
+
+    func setupStackedvaluesDataEntries() -> [ChartDataEntry]
```
Comment (xkampotx) [inline Tests/Charts/HorizontalBarChartTests.swift]: setupStackedValuesDataEntries ;)

Comment (liuxuan30) [reply Tests/Charts/HorizontalBarChartTests.swift]: thanks. fixed

**Comment by liuxuan30** [comment]
one thing to mention is that I couldn't subclass BarChartTests for HorizontalBarCharts, always return failures. so I have to duplicate code for now.  
maybe @alanzeino can give some insights?

**Comment by alanzeino** [comment]
> one thing to mention is that I couldn't subclass BarChartTests for HorizontalBarCharts, always return failures no matter it's in record mode or not, the record mode is even not creating images on my side when subclassing from BarChartTests. so I have to duplicate test code for now.
> maybe @alanzeino can give some insights?

I don't think subclasses test classes is a good idea, you might want to look into the library with some breakpoints to figure out what's going on, for example in `FBSnapshotTestController.m`.

**Comment by liuxuan30** [comment]
@alanzeino noted. thanks

File: Source/Charts/Renderers/HorizontalBarChartRenderer.swift [hunk 2]
```diff
@@ -332,7 +332,7 @@ open class HorizontalBarChartRenderer: BarChartRenderer
             {
                 guard let dataSet = dataSets[dataSetIndex] as? IBarChartDataSet else { continue }
                 
-                if !shouldDrawValues(forDataSet: dataSet) || !(dataSet.isDrawIconsEnabled && dataSet.isVisible)
+                if !shouldDrawValues(forDataSet: dataSet)
```
Comment (jjatie) [inline Source/Charts/Renderers/HorizontalBarChartRenderer.swift]: This should be a guard statement:
```
guard shouldDrawValues(for: dataSet),
    dataSet.isDrawIconsEnabled,
    dataSet.isVisible else 
{
    continue
}
```

Comment (liuxuan30) [reply Source/Charts/Renderers/HorizontalBarChartRenderer.swift]: I have updated all related code to guard statement, have a look please.

**Pull request merged**